### ISSUE TITLE
ssh remote: Failed to connect: Read-only file system (os error 30) and Failed to run command: CROSS_CONTAINER_OPTS="--mount type=bind,src=./target,dst=/app/target" "cross" "build" "--package" "remote_server" "--features" "debug-embed" "--target-dir" "target/remote_server" "--target" "aarch64-unknown-linux-gnu". Please try again.

### ISSUE THREAD

**Issue by opsnull** [zed-industries/zed#19781]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

I can ssh alizj@dev2 in terminal, but can't in zed:

![image](https://github.com/user-attachments/assets/e5d63bdb-1a40-45d4-a2ea-13da870e7c10)

Another problem is cross building failed(my dev2 vm is aarch64 ubuntu 20.04) when I remote ssh alizj@dev2 in zed, the debug log is:

``` shell
 => WARN: InvalidDefaultArgInFrom: Default value for ARG ${CROSS_BASE_IMAGE} results in empty or invalid base image name (line 4)                                                                             0.0s
 => [internal] load .dockerignore                                                                                                                                                                             0.0s
 => => transferring context: 2B                                                                                                                                                                               0.0s
 => resolve image config for docker.io/docker/dockerfile:1                                                                                                                                                    0.0s
 => CACHED docker-image://docker.io/docker/dockerfile:1                                                                                                                                                       0.0s
 => [internal] load metadata for ghcr.io/cross-rs/aarch64-unknown-linux-gnu:main                                                                                                                              1.0s
 => [1/7] FROM ghcr.io/cross-rs/aarch64-unknown-linux-gnu:main@sha256:9808d7b7563a6334042c00555e473d0e0be46a171e3426301030118807280c86                                                                        0.0s
 => [internal] load build context                                                                                                                                                                             0.2s
 => => transferring context: 657.12kB                                                                                                                                                                         0.2s
 => CACHED [2/7] WORKDIR /app                                                                                                                                                                                 0.0s
 => CACHED [3/7] COPY script/install-mold script/                                                                                                                                                             0.0s
 => ERROR [4/7] RUN ./script/install-mold "2.34.0"                                                                                                                                                            0.2s
------                                                                                                                                                                                                             
 > [4/7] RUN ./script/install-mold "2.34.0":
0.121 exec /bin/sh: exec format error
------

 1 warning found (use docker --debug to expand):
 - InvalidDefaultArgInFrom: Default value for ARG ${CROSS_BASE_IMAGE} results in empty or invalid base image name (line 4)
Dockerfile-cross:13
--------------------
  11 |     
  12 |     COPY script/install-mold script/
  13 | >>> RUN ./script/install-mold "2.34.0"
  14 |     COPY script/remote-server script/
  15 |     RUN ./script/remote-server
--------------------
ERROR: failed to solve: process "/bin/sh -c ./script/install-mold \"2.34.0\"" did not complete successfully: exit code: 1

View build details: docker-desktop://dashboard/build/lima-dev2/lima-dev2/brdjyxo5j6pkigcv03fuvaqje
Error: 
   0: could not run container
   1: when building custom image
   2: when building dockerfile
   3: `docker buildx build --platform linux/amd64 --progress auto --label 'org.cross-rs.for-cross-target=aarch64-unknown-linux-gnu' --label 'org.cross-rs.runs-with=x86_64-unknown-linux-gnu' --label 'org.cross-rs.workspace_root=/Users/alizj/go/src/github.com/zed-industries/zed' --tag localhost/cross-rs/cross-custom-zed:aarch64-unknown-linux-gnu-8d728 --build-arg 'CROSS_DEB_ARCH=arm64' --build-arg 'CROSS_BASE_IMAGE=ghcr.io/cross-rs/aarch64-unknown-linux-gnu:main' --file /Users/alizj/go/src/github.com/zed-industries/zed/Dockerfile-cross --output 'type=docker' /Users/alizj/go/src/github.com/zed-industries/zed` failed with exit status: 1

Location:
   src/docker/custom.rs:175

Warning: call to docker failed
Suggestion: is `buildx` available for the container engine?
Note: disable the `buildkit` dependency optionally with `CROSS_CONTAINER_ENGINE_NO_BUILDKIT=1`
[2024-10-26T16:20:45+08:00 ERROR workspace::notifications] Failed to run command: CROSS_CONTAINER_OPTS="--mount type=bind,src=./target,dst=/app/target" "cross" "build" "--package" "remote_server" "--features" "debug-embed" "--target-dir" "target/remote_server" "--target" "aarch64-unknown-linux-gnu"
```

### Environment

Zed: v0.160.0 (Zed Dev d7a277607b17e9875a1aba6351e3acc64db0c1d3)
OS: macOS 15.0.1
Memory: 18 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

<details><summary>Zed.log</summary><pre>
<!-- Click below this line and paste or drag-and-drop your log-->

<!-- Click above this line and paste or drag-and-drop your log--></pre></details>

**Comment by notpeter** [comment]
Cross compiling builds for the remote inside the container is not particular robust.  Others on the team have reported needing to use Docker Desktop to get it working. I experienced errors under podman which required the insane `podman machine ssh mkdir target` workaround to get it working.

Alternatively you set ZED_USE_CACHED_REMOTE_SERVER in your environment and Zed will assume you have already placed a binary (perhaps built previously, or downloaded from GitHub releases) at `~/.zed_server/zed-remote-server-dev-linux-x86_64` on your remote.

**Comment by opsnull** [comment]
@notpeter Thanks for your reply! 

>  use Docker Desktop to get it working

Yes, I also found this issue and successfully cross-compiled a working zed-remote-server binary after using Docker Desktop. Additionally, I fixed a GCC version compatibility issue that occurred during cross-compilation. PR：https://github.com/zed-industries/zed/pull/19910

> Alternatively you set ZED_USE_CACHED_REMOTE_SERVER in your environment 

Aha, this is a good solution. I'm currently using the dev build method, and every time I connect to the server via remote SSH, zed re-cross-compiles the remote server binary, which is very time-consuming and annoying. My workaround solution was to temporarily modify this part of the code logic locally: similarly, it checks if the compiled binary exists locally, and only compiles when it doesn't exist. （see [my note here](https://github.com/opsnull/dotzed/blob/main/zed-note.md#%E6%9E%84%E5%BB%BA)）
I'll try your solution. Thanks again!
